### DISCUSSION TITLE
Does `$anchor` creates new scope for JSON Pointer references?

### DISCUSSION THREAD

**Discussion by flaksp** [json-schema-org/community#258] [Q&A] [answered]
Schema example with blog post that have recursive comments tree:

```json
{
  "$id": "https://example.com/BlogPost",
  "type": "object",
  "properties": {
    "comments": {
      "type": "array",
      "items": { "$ref": "#comment" }
    }
  },
  "$defs": {
    "comment": {
      "$anchor": "comment",
      "type": "object",
      "properties": {
        "text": { "type": "string" },
        "replies": {
          "type": "array",
          "items": { "$ref": "#" }
        }
      }
    }
  }
}
```

The question is at `"items": { "$ref": "#" }` line: will it reference to the `https://example.com/BlogPost#comment` schema or to the `https://example.com/BlogPost` schema? 

It seems like it should reference to the `https://example.com/BlogPost` schema, but I am not sure 🤔

**Comment by handrews** [comment, marked as answer]
It references `https://example.com/BlogPost`.  The base URI is set by `$id` and is `https://example.com/BlogPost` so the `"$ref": "#"` resolves to `"$ref": "https://example.com/BlogPost#"`.  `$anchor` does not change the base URI at all, because base URIs are absolute-URIs (with a scheme and without a fragment).
